saleor/graphql/app/types.py

```diff
@@ -51,6 +54,31 @@ class AppManifestExtension(graphene.ObjectType):
     target = AppExtensionTargetEnum(
         description="Place where extension's iframe will be mounted.", required=True
     )
+    open_as = AppExtensionOpenAsEnum(
+        description="Type of way how app extension will be opened.", required=True
+    )
+
+    @staticmethod
+    def resolve_open_as(root, info):
+        return root.get("open_as") or AppExtensionOpenAs.POPUP
+
+    @staticmethod
+    def resolve_url(root, info):
+        """Return an extension url.
+
+        Apply url stitching when these 3 conditions are met:
+            - url starts with /
+            - openAs == "POPUP"
+            - appUrl is defined
+        """
+        open_as = root.get("open_as", AppExtensionOpenAs.POPUP)
+        app_url = root["app_url"]
+        url = root["url"]
+        if url.startswith("/") and app_url and open_as == AppExtensionOpenAs.POPUP:
+            parsed_url = urlparse(app_url)
+            new_path = urljoin(parsed_url.path, url[1:])
+            return parsed_url._replace(path=new_path).geturl()
+        return url
```

Comment:
Maybe we could move it to `resolvers.py` to easier reuse it in `AppExtension.resolve_url`.